tt_metal/common/assert.hpp

```diff
@@ -156,25 +108,53 @@ template <typename... Ts>
     throw std::runtime_error(trace_message_ss.str());
 }
 
-template <typename... Ts>
-void tt_assert(char const* file, int line, const std::string& assert_type, char const* condition_str, Ts const&... messages) {
-    ::tt::assert::tt_throw(file, line, assert_type, condition_str, messages...);
+[[noreturn]] inline void tt_throw(char const* file, int line, char const* assert_type, char const* condition_str) {
+    tt_throw_impl(file, line, assert_type, condition_str);
+}
+
+template <typename... Args>
+[[noreturn]] void tt_throw(
+    char const* file,
+    int line,
+    char const* assert_type,
+    char const* condition_str,
+    fmt::format_string<Args...> fmt,
+    Args&&... args) {
+    tt_throw_impl(file, line, assert_type, condition_str, fmt, std::forward<Args>(args)...);
+}
```

Comment:
```suggestion
template <typename... Args>
[[noreturn]] void tt_throw(
    char const* file,
    int line,
    char const* assert_type,
    char const* condition_str,
    fmt::format_string<Args...> fmt,
    Args const&... args) {
    tt_throw_impl(file, line, assert_type, condition_str, fmt, args...);
}
```
and below as well. `fmt::format` just needs const references